R/add-tbl.R

```diff
@@ -32,19 +32,23 @@ dm_add_tbl <- function(dm, ..., repair = "unique", quiet = FALSE) {
   old_names <- src_tbls(dm)
   names_list <- repair_table_names(old_names, new_names, repair, quiet)
   # rename old tables in case name repair changed their names
-  dm <- dm_select_tbl_impl(dm, names_list$new_old_names)
 
+  dm <- dm_select_tbl_impl(dm, names_list$new_old_names, needs_repair = FALSE)
   dm_add_tbl_impl(dm, new_tables, names_list$new_names)
 }
 
-repair_table_names <- function(old_names, new_names, repair = "check_unique", quiet = FALSE) {
-  all_names <- tryCatch(
-    vctrs::vec_as_names(c(old_names, new_names), repair = repair, quiet = quiet),
+repair_names_vec <- function(names, repair, quiet) {
+  tryCatch(
+    vctrs::vec_as_names(names, repair = repair, quiet = quiet),
     error = function(e) {
```

Comment:
```suggestion
    vctrs_error_names_must_be_unique = function(e) {
```

and get rid of the check in the body.